### ISSUE TITLE
ec2_asg still terminates an instance before creating a replacement

### ISSUE THREAD

**Issue by 2rs2ts** [ansible/ansible#40337]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME

ec2_asg

##### ANSIBLE VERSION

```
ansible 2.5.0
  config file = /home/admin/.ansible.cfg
  configured module search path = [u'/home/admin/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.13 (default, Nov 24 2017, 17:33:09) [GCC 6.3.0 20170516]
```

##### CONFIGURATION

```
ANSIBLE_SSH_CONTROL_PATH(/home/admin/.ansible.cfg) = %(directory)s/%%C
DEFAULT_CALLBACK_WHITELIST(/home/admin/.ansible.cfg) = ['profile_tasks']
DEFAULT_HOST_LIST(/home/admin/.ansible.cfg) = [u'/home/admin/bin/my-ansible-inventory']
RETRY_FILES_SAVE_PATH(/home/admin/.ansible.cfg) = /tmp
```

##### OS / ENVIRONMENT

`uname -a` output: `Linux mymachine 4.9.0-6-amd64 #1 SMP Debian 4.9.88-1 (2018-04-29) x86_64 GNU/Linux`

##### SUMMARY

The issue in #35993 is not fixed; TL;DR `ec2_asg` creates more instances than it needs to then deletes an old one before it should, not every time but frequently.

##### STEPS TO REPRODUCE

After #35993 was closed, I backported the feature into my project and it seemed to have fixed things. However, after about 3 weeks, the issue returned without me having changed anything about my system. (I suspect it's something on AWS' end that changed.) I asked that the issue be reopened but after 24 days of no response I've decided to file a new issue.

The example from the original issue still reproduces the error, but not 100% of the time... I'd say for me it's about 75% of the time.

##### EXPECTED RESULTS

The expected results from the original issue still apply.

##### ACTUAL RESULTS

The actual results from the original issue still apply.

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/cloud/amazon/ec2_asg.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/ec2_asg.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @garethr @ryansb @s-hertel
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @jillr @tremble
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Thank you very much for your interest in Ansible. Ansible has migrated much of the content into separate repositories to allow for more rapid, independent development. We are closing this issue/PR because this content has been moved to one or more collection repositories.

* lib/ansible/modules/cloud/amazon/ec2_asg.py -> https://galaxy.ansible.com/community/aws

For further information, please see:
https://github.com/ansible/ansibullbot/blob/master/docs/collection_migration.md
